diff.py

```diff
@@ -2050,9 +2078,21 @@ def normalize_stack(row: str, arch: ArchSettings) -> str:
     return re.sub(arch.re_sprel, "addr(sp)", row)
 
 
-def imm_matches_everything(row: str, arch: ArchSettings) -> bool:
-    # (this should probably be arch-specific)
-    return "(." in row
+def imm_matches_everything(field: str, arch: ArchSettings) -> bool:
+    if arch.name == "ppc":
+        if "...data" in field:
+            return True
+
+        field = "".join(field.rsplit("@ha", 1))
+        field = "".join(field.rsplit("@l", 1))
+        field = "".join(field.rsplit("@sda21", 1))
+
+        return re.search(re.compile(r"\A@\d+\Z"), field) != None
```

Comment:
`\A` -> `^`, `\Z` -> `$` and `!= None` -> `is not None` per usual python coding style. `match` or `fullmatch` make more sense than `search` if you want to check that the full string matches something

Reply:
ahh didn't know about fullmatch function

Reply:
If I was to precompile this, where to put it? it's unique to PPC. should I make it global constant? or have a blank field in the other classes, and what to call this field? "re.compile(r"^@\d+$")"

Reply:
I was thinking AsmProcessorPPC, but since that's not available I guess we can skip manually caching the compiled regex and hoping Python's builtin regex compile cache will handle it